### PR TITLE
Abelianization

### PR DESCRIPTION
**Pull request by EgbertRijke**
This pull request formalizes the abelianization of a group. In order to do this:

- I extended the existing infrastructure for Galois connections
- I constructed image-pullback Galois connections for subtypes, subsemigroups and subgroups
- I used these galois connections to give slick proofs that `im f <S> = <im f S>` for any subset of a group
- This implies that any group homomorphism maps commutator subgroups to commutator subgroups
- This was needed to establish functoriality of the abelianization.
- All of this is accompanied by guiding informal texts and informal proofs, diagrams, and  lots of hyperlinks.
- I have built the website locally to make sure that it all looks good.

I already got some helpful comments while this pull request was still a draft. From the comments I gauged that:

- I should fix formatting in all files that were touched by in this PR. Note that this is a recursive condition.
- I should add links to all files touched in this PR. Note that this is a recursive condition too.

So I went on to do this too, because I would get comments about it anyway. While doing this I noticed some other things I wanted to change. Most notably places where we still used `-Prop` as a suffix for propositions made about groups. I went systematically through the library to fix all those instances that were related to semigroups, monoids, commutative monoids, groups and abelian groups. Note that I didn't yet do the same for rings and other structures. That would be too much for me. I did fix some instances of strange capitalization of `Equivalence-Relation` in the middle of a name.

Anyway, that triggered the previous recursive conditions again, so I went through it again :)

### CODE REVIEW

File: src/foundation-core/functoriality-dependent-pair-types.lagda.md [hunk 1]
```diff
@@ -334,12 +334,12 @@ module _
       is-equiv f → is-fiberwise-equiv g → is-equiv (map-Σ D f g)
     is-equiv-map-Σ {f} {g} is-equiv-f is-fiberwise-equiv-g =
       is-equiv-comp-htpy
-        ( map-Σ D f g)
```
Comment (EgbertRijke) [inline src/foundation-core/functoriality-dependent-pair-types.lagda.md:337]: not sure how this one came back to haunt me:P

**Comment by EgbertRijke** [comment]
This pull request is turning into a refactoring PR:/

One thing I noticed was that in order to show that the abelianization is a functor, I need to know that any group homomorphism `f : G -> H` maps the commutator subgroup `[G,G]` to `[H,H]`. The commutator subgroup is the subgroup generated by all the commutators, so one nice way to do this would be to show first that if `(S)` is the subgroup generated by `S`, then `im(f(S)) = (f(S))`, and one nice way to formalize that fact will be via the Galois correspondence `im -| pullback`.

Long story short, there's a bunch of stuff that still has to be done before I have a formalization of abelianization I'm happy with:)

**Comment by fredrik-bakke** [comment]
Lol, sounds like you'll make some nice additions along the way though :) Is your plan to do continue working on this PR for a while longer?

**Comment by EgbertRijke** [comment]
> Lol, sounds like you'll make some nice additions along the way though :) Is your plan to continue work in this PR for a while longer?

My plan is to add it all in here

File: src/category-theory/copresheaf-categories.lagda.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ We record the components of the large category of copresheaves on a precategory.
     obj-Large-Precategory copresheaf-Large-Precategory
 
   hom-set-copresheaf-Large-Category =
```
Comment (EgbertRijke) [inline src/category-theory/copresheaf-categories.lagda.md:76]: ?? There are untyped definitions in our library?

That shouldn't happen.

Comment (fredrik-bakke) [reply src/category-theory/copresheaf-categories.lagda.md]: I figured it was more readable this way. The types are easy to infer and add a lot of bloat to the file.

Comment (fredrik-bakke) [reply src/category-theory/copresheaf-categories.lagda.md]: I.e. I think they are detrimental to readability, and moreover it is big hassle to write them out.

Comment (EgbertRijke) [reply src/category-theory/copresheaf-categories.lagda.md]: Ok, then I’ll write them out myself. I disagree with you that it’s a big hassle and that it’s detrimental to readability. 

1. I discovered this when making the name from hom-set-Large-Poset to hom-Large-Poset. This entry got mistakenly replaced too and it mistakenly type checked because it was lacking a specification. So I’d say it is detrimental to maintainability. 

2. The whole point of naming projections is that when you click on them to see their specification, then you see a relevant formulation of its specification. You seem to be missing that point. 

3. It takes like 30 seconds to specify these definitions. 

Anyway, I’ll add them but please give specifications with your entries next time.

File: src/ring-theory/isomorphisms-rings.lagda.md [hunk 3]
```diff
@@ -483,16 +483,24 @@ module _
...
+                ( Σ ( (x y : type-Ring R) → type-Ring R)
+                    ( λ g →
+                      (x y : type-Ring R) → mul-Ring R x y ＝ g x y))
+                ( equiv-tot
+                  ( λ m →
+                    equiv-Π-equiv-family (λ x → equiv-explicit-implicit-Π) ∘e
+                    equiv-explicit-implicit-Π))
+                ( is-torsorial-Eq-Π
+                  ( λ x m → (y : type-Ring R) → mul-Ring R x y ＝ m y)
+                  ( λ x → is-torsorial-htpy (mul-Ring R x))))
```
Comment (fredrik-bakke) [inline src/ring-theory/isomorphisms-rings.lagda.md:496]: Instead of translating to the explicit case we should have a `is-torsorial-Eq-implicit-Π`

File: src/order-theory/similarity-of-elements-large-preorders.lagda.md [hunk 4]
```diff
@@ -94,3 +98,16 @@ module _
...
+
+```agda
+module _
+  {α : Level → Level} {β : Level → Level → Level} (P : Large-Preorder α β)
+  where
+
+  sim-eq-Large-Preorder :
+    {l : Level} {x y : type-Large-Preorder P l} →
+    x ＝ y → sim-Large-Preorder P x y
+  sim-eq-Large-Preorder refl = refl-sim-Large-Preorder P _
```
Comment (fredrik-bakke) [inline src/order-theory/similarity-of-elements-large-preorders.lagda.md:112]: I find the following form preferable. The `_` confuses me although it becomes immediately clear once I read the rest of the definition.
```suggestion
  sim-eq-Large-Preorder {x = x} refl = refl-sim-Large-Preorder P x
```

File: src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+     | |        | |
+  LF | |     LG | |
+     |⊣| UF     |⊣| UG
+     | |        | |
+     ∨ |   LJ   ∨ |
+        ------->  |
+      Q     ⊥    V.
+        <-------
+           UJ
+```
```
Comment (fredrik-bakke) [inline src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md:39]: My eyes! 😵‍💫

File: src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+between large posets. We say that the diagram commutes if there is a
```
Comment (fredrik-bakke) [inline src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md:41]:
```suggestion
between [large posets](order-theory.large-posets.md). We say that the diagram **commutes** if there is a
```

File: src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+[similarity](order-theory.similarity-of-order-preserving-maps-large-posets.md)
+`LJ ∘ LF ≈ LG ∘ LI`. This condition is equivalent the condition that there is a
```
Comment (fredrik-bakke) [inline src/order-theory/commuting-squares-of-galois-connections-large-posets.lagda.md:43]:
```suggestion
`LJ ∘ LF ≈ LG ∘ LI`. This condition is [equivalent](foundation.logical-equivalences.md) the condition that there is a
```

File: src/order-theory/lower-sets-large-posets.lagda.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+open import order-theory.large-posets
+open import order-theory.large-subposets
+```
+
+</details>
+
+## Idea
+
+An **lower set** or **downwards closed set** in a
```
Comment (fredrik-bakke) [inline src/order-theory/lower-sets-large-posets.lagda.md:20]:
```suggestion
A **lower set** or **downwards closed set** in a
```

File: src/order-theory/lower-sets-large-posets.lagda.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
+# Lower sets of large posets
```
Comment (fredrik-bakke) [inline src/order-theory/lower-sets-large-posets.lagda.md:1]: This phrase reads a little strangely to me. Why not phrase this file as lower **sub**sets?

File: src/order-theory/upper-sets-large-posets.lagda.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
+# Upper sets of large posets
```
Comment (fredrik-bakke) [inline src/order-theory/upper-sets-large-posets.lagda.md:1]: Same here. Wouldn't it be better to phrase it as an upper **sub**set?

**Review by fredrik-bakke — COMMENTED** [review]
I have 10 files left to review, but I'm taking a break now. I'm loving some of the new files. They are very well written!

File: src/group-theory/subgroups-generated-by-elements-groups.lagda.md [hunk 11]
```diff
@@ -225,55 +233,145 @@ module _
...
+### The image of the subgroup `⟨x⟩` generated by `x` under a group homomorphism `f` is the subgroup `⟨f(x)⟩` generated by the element `f(x)`
+
+There are three ways to state this fact:
+
+1. The image of the subgroup `⟨x⟩` under the group homomorphism `f` has the same
+   elements as the subgroup `⟨f(x)⟩`.
+2. The subgroup `⟨f(x)⟩` satisfies the universal property of the image of the
+   group `⟨x⟩` under the group homomorphism `f`.
+3. The image of the subgroup `⟨x⟩` under the group homomorphism `f` satisfies
+   the universal property of the subgroup generated by the element `f x`.
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/subgroups-generated-by-elements-groups.lagda.md:320]: Yeah, I want a separate treatment for a single element. It's definitely not an oversight. The difference between a single elements `x : type-Group G` and a family of elements `x : unit -> type-Group G` indexed by the unit type would be super annoying to deal with, if that's what you had in mind.

[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/subgroups-generated-by-elements-groups.lagda.md:320]: No, that is not what I had in mind. What I had in mind is that you define the results for the treatment of a single element as corollaries of the results for a family of elements via `point`.

File: src/foundation/pullbacks-subtypes.lagda.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+## Idea
+
+Consider a [subtype](foundation-core.subtypes.md) `T` of a type `B` and a map
+`f : A → B`. Then the **pullback subtype** `pullback f T` of `A` is defined to
+be `T ∘ f`. This fits in a [pullback diagram](foundation-core.pullbacks.md)
+
+```text
+                 π₂
+  pullback f T -----> T
+       |              |
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/foundation/pullbacks-subtypes.lagda.md:33]: Sure, great suggestion. I'll implement the pullback corners throughout.

File: src/group-theory/subgroups-generated-by-elements-groups.lagda.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/subgroups-generated-by-elements-groups.lagda.md:320]: Or the other way around, I suppose.

File: src/group-theory/images-of-semigroup-homomorphisms.lagda.md [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+More generally, the **image of a subsemigroup** `S` under a semigroup
+homomorphism `f : G → H` is the subsemigroup consisting of all the elements in
+the image of the underlying subset of `S` under the underlying map of `f`. Since
+the image of a subsemigroup satisfies the following adjoint relation
+
+```text
+  (im f S ⊆ T) ↔ (S ⊆ T ∘ f)
+```
+
+it follows that we obtain a
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/images-of-semigroup-homomorphisms.lagda.md:51]: My bad

File: src/order-theory/lower-sets-large-posets.lagda.md (hunk 9 above)
Comment (EgbertRijke) [reply src/order-theory/lower-sets-large-posets.lagda.md:1]: That's what they are called:

https://en.wikipedia.org/wiki/Upper_set

🤷‍♂️

File: src/ring-theory/isomorphisms-rings.lagda.md (hunk 3 above)
Comment (EgbertRijke) [reply src/ring-theory/isomorphisms-rings.lagda.md:496]: Oh yeah absolutely!

File: src/group-theory/subgroups-generated-by-elements-groups.lagda.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/subgroups-generated-by-elements-groups.lagda.md:320]: Ah ok, I suppose it could be done that way. I'm gonna put my priority to finishing the PR though.

As a side benefit of my dumb approach is that now we have some extra facts about singleton subsets in the library, lol:)

File: src/order-theory/similarity-of-elements-large-preorders.lagda.md (hunk 4 above)
Comment (EgbertRijke) [reply src/order-theory/similarity-of-elements-large-preorders.lagda.md:112]: Perhaps you're right. I don't love the `{x = x}  notation though. It seems so heavy compared to a simple `_`

Comment (EgbertRijke) [reply src/order-theory/similarity-of-elements-large-preorders.lagda.md:112]: On the other hand, we have several places in the library where we don't declare implicit variables and use underscores instead. I think it is not uncommon and I wouldn't want to make a policy out of declaring our implicit variables always, in favor of using less under scores. It seems that this is (and should be) up to the eye of the beholder, which way to go.

File: src/order-theory/lower-sets-large-posets.lagda.md (hunk 9 above)
Comment (fredrik-bakke) [reply src/order-theory/lower-sets-large-posets.lagda.md:1]: 🤷‍♂️

Comment (fredrik-bakke) [reply src/order-theory/lower-sets-large-posets.lagda.md:1]: Then may we at least use the preposition `in`, to not suggest it is somehow a "lower set" in the category of posets?

Comment (fredrik-bakke) [reply src/order-theory/lower-sets-large-posets.lagda.md:1]: (It should be permissible to use `in`, as they do that too in the wiki article)

File: src/order-theory/similarity-of-elements-large-preorders.lagda.md (hunk 4 above)
Comment (EgbertRijke) [reply src/order-theory/similarity-of-elements-large-preorders.lagda.md:112]: There is no objective evaluation of “what complicates the definition the least”. For instance, I’m happy to with underscores as a way to tell Agda to figure out itself how to fill out that value, but less happy to use underscores in partially filled out functions with infix notation, but your evaluation function is different. What I’m saying is that it’s up to the author as long as the code looks reasonable.

File: src/group-theory/kernels.lagda.md [hunk 14]
```diff
@@ -106,15 +106,15 @@ module _
...
         ( mul-Group' H (map-hom-Group G H f (inv-Group G g)))
-        ( ( preserves-mul-hom-Group G H f g (pr1 h)) ∙
+        ( ( preserves-mul-hom-Group G H f) ∙
           ( ( ap (mul-Group H (map-hom-Group G H f g)) (pr2 h)) ∙
             ( right-unit-law-mul-Group H (map-hom-Group G H f g))))) ∙
       ( ( ap
           ( mul-Group H (map-hom-Group G H f g))
-          ( preserves-inv-hom-Group G H f g)) ∙
+          ( preserves-inv-hom-Group G H f)) ∙
         ( right-inverse-law-mul-Group H (map-hom-Group G H f g))))
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/kernels.lagda.md:118]: Oh, wait, this is wrong. Sorry! I pasted the wrong code.

File: src/group-theory/homomorphisms-generated-subgroups.lagda.md [hunk 15]
```diff
@@ -59,49 +59,44 @@ module _
...
+        ( ( ap
+            ( mul-Group G'
+              ( map-hom-Group
+                ( group-Subgroup G (subgroup-subset-Group G S))
+                ( G')
+                ( f)
+                ( pair
+                  ( ev-formal-combination-subset-Group G S (unit-list (s , x)))
+                  ( unit-trunc-Prop (unit-list (s , x) , refl)))))
+            ( distributivity-hom-group-ev-formal-combination f l))))
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/homomorphisms-generated-subgroups.lagda.md]: I systematically went through all the files in group theory to change the formatting style of all concatenated equalities that I could find.

[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/homomorphisms-generated-subgroups.lagda.md]: Oh, wow, thanks for doing that!

**Comment by EgbertRijke** [comment]
I have a request for this PR.

I worked really hard on this one for actually quite a few weeks (like, at least 3 or 4 I think), and it was not so easy to map out the formalization path, build all the required infrastructure, and explain everything nicely. Furthermore, the whole thing is delivered nicely refactored, and there are no long constructions.

All of that was a lot of work, so my request is whether I can be the sole author of this pull request:)

**Comment by fredrik-bakke** [comment]
> I have a request for this PR.
> 
> I worked really hard on this one for actually quite a few weeks (like, at least 3 or 4 I think), and it was not so easy to map out the formalization path, build all the required infrastructure, and explain everything nicely. Furthermore, the whole thing is delivered nicely refactored, and there are no long constructions.
> 
> All of that was a lot of work, so my request is whether I can be the sole author of this pull request:)

Of course! Maybe it is good we discuss guidelines for this? If I'm not mistaken, there's an easy solution that allows us to write suggestions and still not be attributed as co-authors if those suggestions are merged, and that is to simply edit the final commit message and remove the `Co-authored by` lines.

**Comment by fredrik-bakke** [comment]
As a first proposal, we can say that giving `suggestion`s does not merit co-authorship.

**Comment by fredrik-bakke** [comment]
While on topic, I noticed that 1lab uses "chore" labels/keywords to hide certain PRs from the history on the webpage, and I wonder if that would be good to instate in this repo as well, so not to dirty the history shown on the website with simple formatting changes. I'm sure Vojtech will be happy with this change as well, as it forces us to be more structured in our PRs ;)

**Comment by fredrik-bakke** [comment]
> I have a request for this PR.
> 
> I worked really hard on this one for actually quite a few weeks (like, at least 3 or 4 I think), and it was not so easy to map out the formalization path, build all the required infrastructure, and explain everything nicely. Furthermore, the whole thing is delivered nicely refactored, and there are no long constructions.
> 
> All of that was a lot of work, so my request is whether I can be the sole author of this pull request:)

Does this warrant the `milestone` label btw? :)

File: src/order-theory/lower-sets-large-posets.lagda.md [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+  where
+
+  is-lower-set-Large-Subposet : UUω
+  is-lower-set-Large-Subposet =
+    {l1 l2 : Level} (x : type-Large-Poset P l1) (y : type-Large-Poset P l2) →
+    leq-Large-Poset P y x →
+    is-in-Large-Subposet P S x → is-in-Large-Subposet P S y
+```
+
+### Lower sets of a large poset
```
Comment (fredrik-bakke) [inline src/order-theory/lower-sets-large-posets.lagda.md:46]:
```suggestion
### Lower sets of a large poset
```
```suggestion
### Lower sets in a large poset
```

File: src/order-theory/order-preserving-maps-large-posets.lagda.md [hunk 17]
```diff
@@ -35,27 +62,196 @@ module _
...
+      ( large-preorder-Large-Poset Q)
+      ( f)
+```
+
+## See also
+
+- [Order preserving maps](order-theory.order-preserving-maps-large-posets.md)
+  between large posets
+- [Similarity](order-theory.similarity-of-order-preserving-maps-large-posets.md)
+  of order preserving maps between large posets
```
Comment (fredrik-bakke) [inline src/order-theory/order-preserving-maps-large-posets.lagda.md:257]: See also functors between large categories

File: src/order-theory/order-preserving-maps-large-preorders.lagda.md [hunk 18]
```diff
@@ -35,21 +62,155 @@ module _
...
+  pr2 (preserves-sim-hom-Large-Preorder x y (s , t)) =
+    preserves-order-hom-Large-Preorder f y x t
+```
+
+## See also
+
+- [Order preserving maps](order-theory.order-preserving-maps-large-posets.md)
+  between large posets
+- [Similarity](order-theory.similarity-of-order-preserving-maps-large-preorders.md)
+  of order preserving maps between large preorders
```
Comment (fredrik-bakke) [inline src/order-theory/order-preserving-maps-large-preorders.lagda.md:216]: See also functors between large precatgories

File: src/order-theory/principal-lower-sets-large-posets.lagda.md [hunk 19]
```diff
@@ -0,0 +1,131 @@
+# Principal lower sets of large posets
```
Comment (fredrik-bakke) [inline src/order-theory/principal-lower-sets-large-posets.lagda.md:1]:
```suggestion
# Principal lower sets in large posets
```

File: src/order-theory/principal-lower-sets-large-posets.lagda.md [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+      ( principal-lower-set-element-Large-Poset P y)
+  pr1
+    ( has-same-elements-principal-lower-set-element-sim-Large-Poset (H , K) z) =
+    transitive-leq-Large-Poset P z x y H
+  pr2
+    ( has-same-elements-principal-lower-set-element-sim-Large-Poset (H , K) z) =
+    transitive-leq-Large-Poset P z y x K
+```
+
+### For two elements `x` and `y` of a large poset the same universe level, if the principal lower sets `↓{x}` and `↓{y}` have the same elements, then `x` and `y` are equal
```
Comment (fredrik-bakke) [inline src/order-theory/principal-lower-sets-large-posets.lagda.md:114]:
```suggestion
### For two elements `x` and `y` of a large poset at the same universe level, if the principal lower sets `↓{x}` and `↓{y}` have the same elements, then `x` and `y` are equal
```

File: src/order-theory/principal-upper-sets-large-posets.lagda.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
+# Principal upper sets of large posets
```
Comment (fredrik-bakke) [inline src/order-theory/principal-upper-sets-large-posets.lagda.md:1]:
```suggestion
# Principal upper sets in large posets
```

File: src/order-theory/principal-upper-sets-large-posets.lagda.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+    ( has-same-elements-principal-upper-set-element-sim-Large-Poset (H , K) z)
+    ( p) =
+    transitive-leq-Large-Poset P y x z p K
+  pr2
+    ( has-same-elements-principal-upper-set-element-sim-Large-Poset (H , K) z)
+    ( q) =
+    transitive-leq-Large-Poset P x y z q H
+```
+
+### For two elements `x` and `y` of a large poset the same universe level, if the principal upper sets `↑{x}` and `↑{y}` have the same elements, then `x` and `y` are equal
```
Comment (fredrik-bakke) [inline src/order-theory/principal-upper-sets-large-posets.lagda.md:116]:
```suggestion
### For two elements `x` and `y` of a large poset at the same universe level, if the principal upper sets `↑{x}` and `↑{y}` have the same elements, then `x` and `y` are equal
```

File: src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md [hunk 23]
```diff
@@ -30,7 +30,7 @@ pushout of `S` if and only if the square
 
 ```text
   Y^X -----> Y^B
-   |          |
+   | ⌟        |
```
Comment (fredrik-bakke) [inline src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md:33]: Okay, so here's a distinction I usually make in my own writing. I usually only use pullback corners in diagrams that are established to be pullback diagrams. When the question is posed of a square to be a pullback, rather than taken as a fact, I don't usually include it. Indeed, the square drawn here is not generally a pullback square for arbitrary diagrams of this form.

I'm curious to hear if you have a differing opinion. 

```suggestion
   |          |
```

**Review by fredrik-bakke — APPROVED** [review]
Absolutely crazy amount of neat theory developed in this PR. Well done, Egbert, and congrats with being done!

File: src/group-theory/loop-groups-sets.lagda.md [hunk 24]
```diff
@@ -213,29 +213,29 @@ module _
...
-          ( eq-pair-Σ p (eq-is-prop (is-prop-is-set (type-Set X))))
-          ( eq-pair-Σ q (eq-is-prop (is-prop-is-set (type-Set X))))
-          ( eq-is-prop is-prop-type-trunc-Prop)
-          ( eq-is-prop is-prop-type-trunc-Prop)))
+    ( ap
+      ( λ w →
+        eq-pair-Σ
+          ( ( eq-pair-Σ p (eq-is-prop (is-prop-is-set (pr1 X)))) ∙
+            ( eq-pair-Σ q (eq-is-prop (is-prop-is-set (pr1 X)))))
+          ( w))
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/loop-groups-sets.lagda.md:232]: The indentation levels are off in this suggestion:)

File: src/group-theory/loop-groups-sets.lagda.md [hunk 25]
```diff
@@ -213,29 +213,29 @@ module _
...
         ( eq-is-prop (is-prop-is-set (type-Set X))))
       ( eq-is-prop is-prop-type-trunc-Prop)
-  pr2 hom-abstract-automorphism-group-loop-group-Set p q =
+  pr2 hom-abstract-automorphism-group-loop-group-Set {p} {q} =
     ( ap
       ( λ r → eq-pair-Σ r (eq-is-prop is-prop-type-trunc-Prop))
       ( ( ap
-        ( λ w → eq-pair-Σ (p ∙ q) w)
-        ( eq-is-prop (is-trunc-Id (is-prop-is-set (type-Set X) _ _)))) ∙
+          ( λ w → eq-pair-Σ (p ∙ q) w)
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/loop-groups-sets.lagda.md:220]: So I really must check all the formatting in all the old code of all files that I touch, hehe. It was not a joke:)

[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/group-theory/loop-groups-sets.lagda.md:220]: Nono. Sorry, feel free to ignore the suggestion.

[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/loop-groups-sets.lagda.md:220]: Haha no I didn't mean that:) It has already been observed, so now it must be edited:) 

Please don't be sorry, such suggestions only make the library better!

File: src/group-theory/abelianization-groups.lagda.md [hunk 26]
```diff
@@ -0,0 +1,460 @@
...
+## External links
+
+- [Abelianization](https://groupprops.subwiki.org/wiki/Abelianization) at
+  Groupprops
+- [Abelianization](https://ncatlab.org/nlab/show/abelianization) at $n$lab
+- [Abelianization](https://en.wikipedia.org/wiki/Commutator_subgroup#Abelianization)
+  at Wikipedia
+- [Abelianization](https://mathworld.wolfram.com/Abelianization.html) at Wolfram
+  Mathworld
+- [Commutator subgroup](https://www.wikidata.org/entity/Q522216) at Wikidata
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/group-theory/abelianization-groups.lagda.md:460]: Lol, the wikidata entry only says that the abelianization is the quotient with the commutator subgroup. If I would have been happy with only that definition this PR could have been done weeks ago:)

**Comment by VojtechStep** [comment]
> While on topic, I noticed that 1lab uses "chore" labels/keywords to hide certain PRs from the history on the webpage, and I wonder if that would be good to instate in this repo as well, so not to dirty the history shown on the website with simple formatting changes. I'm sure Vojtech will be happy with this change as well, as it forces us to be more structured in our PRs ;)

I did partially sneak this feature through when I did the git integration, so commits starting with `chore:` don't contribute to the author list, but since we weren't using it I didn't add the same filter for other stats like Recent commits. I can add that if we want to start using the labels.

Unfortunately that doesn't solve the problem with co-authorship via GH suggestion, because we have no control over those commits, nor how they are interpreted when merging, because that all happens through GitHub's UI.

File: src/foundation/reflecting-maps-equivalence-relations.lagda.md [hunk 27]
```diff
@@ -85,17 +85,17 @@ module _
   (q : A → type-Set B)
   where
 
-  reflects-Equivalence-Relation-is-surjective-and-effective :
+  reflects-equivalence-relation-is-surjective-and-effective :
```
[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/foundation/reflecting-maps-equivalence-relations.lagda.md:88]: It is not at all true that we use `equivalence-relation` everywhere, but why not, I'll fix that for you.

[the remark this answers is not in the document]
Comment (EgbertRijke) [reply src/foundation/reflecting-maps-equivalence-relations.lagda.md:88]: This suggestion caused many hundreds of changes accross the library by the way. After doing it, I realized that it should have done in a separate PR.

File: src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md (hunk 23 above)
Comment (EgbertRijke) [reply src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md:33]: If you assume something is an equivalence, you also use the symbol for equivalences. So if you assume something is a pullback square, then you use the symbol for a pullback square. I don't see why not.

**Comment by fredrik-bakke** [comment]
This PR also merged erroneous renamings like `eq-relate` -> `eq-relationate`. I'm fixing it in #935.

File: src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md (hunk 23 above)
Comment (fredrik-bakke) [reply src/synthetic-homotopy-theory/pullback-property-pushouts.lagda.md:33]: I don't usually write "assume that `f : A ≃ B` is an equivalence", or "assume that the following is an equivalence: `f : A ≃ B`".

**Comment by fredrik-bakke** [comment]
It also renamed `Equivalence-Relation` to `equivalence-relation` which is another mistake. Either because it was unintentional, or since `Decidable-equivalence-relation` was not renamed to `decicable-equivalence-relation` and `Relation` was not renamed to `relation`.

**Comment by EgbertRijke** [comment]
Thank you for fixing those!

**Comment by fredrik-bakke** [comment]
I didn't fix the last one as I was not sure what your intention was
